src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -242,9 +242,7 @@ public final synchronized void shutdown() throws InterruptedException
          LOGGER.info("{} - is closing down.", poolName);
          logPoolState("Before closing ");
 
-         connectionBag.close();
-         softEvictConnections();
-         addConnectionExecutor.shutdownNow();
+         addConnectionExecutor.shutdown();
```

Comment:
IMO it is bad to throw exception to divert program flow.
bag add() is innocent and should not throw exception.
when shutdown() closes executor, there wont be any 'new' connection added.
those already in queue (to be added) or in bag will be closed in eviction later in shutdown